### PR TITLE
Add support for sorting records

### PR DESCRIPTION
**Pull request by edwardhorsford**
Adds support for sorting records.

As a byproduct I've also moved the search input to the filter bar - which matches Manage.

Screenshot 2020-11-03 at 11 24 01

### CODE REVIEW

File: app/assets/sass/components/_all-records-search.scss [hunk 1]
```diff
@@ -17,4 +17,91 @@
...
+
+.gem-c-search__submit {
+  border: 0;
+  cursor: pointer;
+  border-radius: 0;
+  position: relative;
+  padding: 0;
+  width: $input-size;
+  height: $input-size;
+  background-image: url("/public/images/search-button.png");
```
Comment (jesseyuen) [inline app/assets/sass/components/_all-records-search.scss:37]: I feel like an inline <svg> icon with the appropriate A11Y support (`aria-hidden="true" focusable="false"`) would be a better option than a png background image? Fine for the prototype though.

File: app/assets/sass/components/_all-records-search.scss [hunk 2]
```diff
@@ -17,4 +17,91 @@
...
+  background-repeat: no-repeat;
+  background-position: 2px 50%;
+  text-indent: -5000px;
```
Comment (jesseyuen) [inline app/assets/sass/components/_all-records-search.scss:40]: Wrapping the button text in a `<span class="govuk-visually-hidden">Search</span>` would remove the need for the extreme text-indent value.

**Review by jesseyuen — APPROVED** [review]

File: app/assets/sass/components/_all-records-search.scss (hunk 1 above)
Comment (edwardhorsford) [reply app/assets/sass/components/_all-records-search.scss:37]: This stuff was copied from the govuk publishing app. I've now commented this bit out though.

File: app/assets/sass/components/_all-records-search.scss (hunk 2 above)
Comment (edwardhorsford) [reply app/assets/sass/components/_all-records-search.scss:40]: This stuff was copied from the govuk publishing app. I've now commented this bit out though.

**Pull request merged**